feat(update-chart-dependency): implement action (MAPCO-8106) (#17)
MapColonies/javascript-github-actions: actions/update-chart-dependency/README.md

# Update Helm Chart Dependencies GitHub Action

> Generated by Copilot

This GitHub Action updates the version of a specific dependency in the `Chart.yaml` (and optionally `helmfile.yaml`) for one or more Helm charts in your Kubernetes project, creates a branch with the changes, and opens a pull request.

## Features

- Accepts a dependency service name and version as inputs.
- Finds all chart directories (optionally filtered by prefix).
- Updates the dependency version in each chart's `Chart.yaml` and `helmfile.yaml` if present.
- Creates a new branch from a specified base branch.
- Commits all changes and opens a PR with a summary of updated charts.
- Uses only Node.js built-in modules and the official GitHub Actions toolkit.
- All commits are annotated with `Generated by Copilot`.

## Inputs

| Name           | Description                                                                                              | Required | Default  |
| -------------- | -------------------------------------------------------------------------------------------------------- | -------- | -------- |

> | `service-name` | Name of the dependency to update in Chart.yaml/helmfile.yaml.                                            | true     |          |

Author: You are correct, I'll change it.

> | `version`      | New version to set for the dependency.                                                                   | true     |          |

Author: Yes, but the version is mandatory because it needs to be supplied to the action, otherwise we won't know what's the new version (you still need to pass it).

| `github-token` | GitHub token for authentication.                                                                         | true     |         |
| `chart-prefix` | Prefix to filter chart directories. Only charts whose directory starts with this prefix will be updated. | false    |         |
| `branch`       | Branch to base the PR on (e.g. `main`).                                                                  | false    | `main`  |

## Usage

```yaml
- name: Update Helm Chart Dependencies
  uses: map-colonies/update-chart-dependency@v1
  with:
    service-name: 'my-dependency'
    version: '1.2.3'
    github-token: ${{ secrets.GITHUB_TOKEN }}
    chart-prefix: 'myservice-'

>     branch: 'main'

Author: Changed.

```

## Notes

- The action updates only the version of the specified dependency in each chart's `Chart.yaml` and `helmfile.yaml`.

> - The action opens a single PR for all updated charts.

Author: @netanelC Wrote a comment saying that we will open one PR for each domain (prefix). This means IMO that you will call the workflow for each domain if you want to update more than one (or call without any prefix).
If we want to change this, we will need to add another optional input of a prefix list to indicate how to seperate the PRs.

> - The action opens a single PR for all updated charts.

Reviewer: this is bad, but not my choice.
its very brittle.

> - The action opens a single PR for all updated charts.

Reviewer 2: Currently, we won't add this feature. We will keep it clean without the prefix.

- If no charts require updating, no PR will be opened.

> - All commits are annotated with `Generated by Copilot`.

Author: Changed.

# Update Helm Chart Dependencies GitHub Action

> This GitHub Action updates the version of a specific dependency in the `Chart` and `helmfile` yaml files for one or more directories in your repository, creates a branch with the changes, and opens a pull request.

Reviewer 2: Revise this README based on the PR issues (remote repo instead of "your" repo and etc)

Author: Updated.

## Action Operation Diagram

```mermaid
flowchart TD
    A["Tag push in source repo (with Helm chart)"] -->|"GitHub Action triggered"| B

>     B["Checkout source repo"] --> C["Extract chart name & version from tag"]

Reviewer 2: I think it is unnecessary and not related very much to this flow

Author: Removed.

    C --> D["Clone target repo (with multiple chart / helmfile dirs)"]
    D --> E["Scan all directories for Chart /helmfile yaml files"]
    E --> F["Update dependency version for matching chart / helmfile"]
    F --> G{"Any changes?"}
    G -- No --> H["Exit: No update needed"]
    G -- Yes --> I["Create branch, commit changes"]
    I --> J["Open Pull Request to target repo"]
    J --> K["Review & merge"]

    style A fill:#f9f,stroke:#333,stroke-width:2px
    style J fill:#bbf,stroke:#333,stroke-width:2px
```

## Features

- Accepts a dependency service name and version as inputs.

> - Finds all chart directories (optionally filtered by prefix).

Reviewer 2: ```suggestion
- Finds all chart directories.
```

> - Updates the dependency version in each chart's `Chart` and `helmfile` yaml files if present.

Reviewer 2: ```suggestion
- Updates the version of the specified chart in all Chart.yaml and helmfile.yaml files where it is listed as a dependency.
```

# Update Helm Chart Dependencies GitHub Action

This GitHub Action updates the version of a specific dependency in the `Chart` and `helmfile` yaml files for one or more directories in a remote repository. For each it creates or updates a dedicated branch with the chart changes and opens or updates a pull request.

## Features

- Accepts a dependency service name and version as inputs.
- Finds all chart directories in a remote repository.
- Updates the version of the specified chart in all Chart.yaml and helmfile.yaml files where it is listed as a dependency.

> - Creates or updates a dedicated branch for each change and opens ot updates a PR with a summary of the update details.

Reviewer 2: ```suggestion
- Creates or updates a dedicated branch for each change and opens or updates a PR with a summary of the update details.
```
